lib/src/services/messaging_service.dart

```diff
@@ -0,0 +1,245 @@
+/// SPDX-License-Identifier: AGPL-3.0-or-later
+import 'package:archethic_lib_dart/src/model/address.dart';
+import 'package:archethic_lib_dart/src/model/crypto/key_pair.dart';
+import 'package:archethic_lib_dart/src/model/keychain.dart';
+import 'package:archethic_lib_dart/src/model/messaging/ae_discussion.dart';
+import 'package:archethic_lib_dart/src/model/messaging/ae_message.dart';
+import 'package:archethic_lib_dart/src/model/transaction.dart';
+import 'package:archethic_lib_dart/src/services/api_service.dart';
+import 'package:archethic_lib_dart/src/utils/messenger_util.dart';
+
+class MessagingService with MessengerMixin {
+  MessagingService({
+    this.logsActivation = true,
+  });
+
+  /// [logsActivation] manage log activation
+  final bool logsActivation;
+
+  /// Create a new discussion
+  /// @param{Keychain} keychain used to send transaction to BC
+  /// @param{ApiService} API with blockchain
+  /// @param{List<String>} List of ALL members of the discussion (public key)
+  /// @param{String} name of the discussion
+  /// @param{List<String>} List of ALL admins of the discussion (public key)
+  /// @param{String} Address of the admin who want to add members (this address will provision the SC's chain to manage fees)
+  /// @param{String} Service name in the current keychain (= admin)
+  Future<Transaction> createDiscussion({
+    required Keychain keychain,
+    required ApiService apiService,
+    required List<String> membersPubKey,
+    required String discussionName,
+    required List<String> adminsPubKey,
+    required String adminAddress,
+    required String serviceName,
+  }) async {
+    return createTransactionSC(
+      keychain: keychain,
+      apiService: apiService,
+      membersPubKey: membersPubKey,
+      discussionName: discussionName,
+      adminsPubKey: adminsPubKey,
+      adminAddress: adminAddress,
+      serviceName: serviceName,
+    );
+  }
+
+  /// Add member(s) in existing discussion
+  /// You should provide the complete list of member (with new member(s))
+  /// @param{Keychain} keychain used to send transaction to BC
+  /// @param{ApiService} API with blockchain
+  /// @param{String} Smart contract's address
+  /// @param{List<String>} List of ALL members of the discussion (public key)
+  /// @param{String} name of the discussion
+  /// @param{List<String>} List of ALL admins of the discussion (public key)
+  /// @param{String} Address of the admin who want to add members (this address will provision the SC's chain to manage fees)
+  /// @param{String} Service name in the current keychain (= admin)
+  Future<Transaction> addMembers({
```

Comment:
The updateDiscussion method isn't it the correct method to use instead of the addMembers / removeMembers one ?
For example, if in a discussion we have two members (M1 and M2) and we call the removeMembers method with four members (M1, M2, M3 and M4), then we will have 4 members in the discussion which is very weird don't you think ?

Reply:
Yes. for the moment, there is no control in these method. So we could add controls to check. But i confess i created addMembers / removeMembers because of our discussion but just updateDiscussion could be necessary

Reply:
Do you think that in the future we will need to use these methods instead of the updateDiscussion one ? If I think about that for 5 min, we are going to use the update method in the front.

Reply:
agree. i remove these methods